src/components/MyAccount/EditPopUpWindow.tsx

```diff
@@ -0,0 +1,218 @@
+import { AxiosResponse } from "axios";
+import React, { useState } from "react";
+import { useDispatch } from "react-redux";
+import { updateUser, validPassword } from "../Login/helpers/userRequests";
+import validatePwd from "../Login/helpers/validatePwd";
+import EditInputGroup from "./EditInputGroup";
+
+const EditPopUpWindow = ({
+  onClose,
+  isOpen,
+  title,
+}: {
+  onClose: () => void;
+  isOpen: boolean;
+  title: string;
+}) => {
+  const [newFirstName, setNewFirstName] = useState("");
+  const [newLastName, setNewLastName] = useState("");
+  const [oldPsw, setOldPsw] = useState("");
+  const [newPsw, setNewPsw] = useState("");
+  const [confirmPsw, setConfirmPsw] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+  const [weakPasswordMsg, setWeakPasswordMsg] = useState("");
+  const [incorrectPswMsg, setIncorrectPswMsg] = useState("");
+  const currentUserId = useStoreSelector(userData).userId;
+  const toast = useToast();
+  const dispatch = useDispatch();
+
+  const isBlank = (): boolean => {
+    switch (title) {
+      case "Change Your Name":
+        return newFirstName === "" || newLastName == "";
+      case "Change Your Password":
+        return oldPsw === "" || newPsw === "" || confirmPsw === "";
+      default:
+        return true;
+    }
+  };
+
+  const enterPress = (event: any) => {
+    if (event.key === "Enter" && !isBlank()) {
+      event.preventDefault();
+      handleApply();
+    }
+  };
+
+  const failMessage = () => {
+    toast({
+      title: "Operation failed, please try again",
+      status: "error",
+      isClosable: true,
+      position: "bottom",
+    });
+  };
+  const successMessage = () => {
+    toast({
+      title: "Operation successful!",
+      status: "success",
+      isClosable: true,
+      position: "bottom",
+    });
+  };
+
+  const handleApply = async () => {
+    switch (title) {
+      case "Change Your Name": {
+        const res: AxiosResponse = await updateUser({
+          userId: currentUserId,
+          firstName: newFirstName,
+          lastName: newLastName,
+        });
+
+        if (res.status !== 200) {
+          failMessage();
+          return;
+        }
+        successMessage();
+        const data = res.data;
+        localStorage.setItem("UserInfo", JSON.stringify(data));
+        dispatch(
+          updateUserInfo({
+            userId: data._id,
+            email: data.email,
+            firstName: data.firstName,
+            lastName: data.lastName,
+          })
+        );
+        onClose();
+        break;
+      }
+      case "Change Your Password": {
+        validatePwd(newPsw, confirmPsw, setWeakPasswordMsg, setErrorMessage);
+        const response: AxiosResponse = await validPassword({
+          userId: currentUserId,
+          password: oldPsw,
+        });
+        if (response.status !== 200) {
+          failMessage();
+          return;
+        }
+        const result = response.data;
+        result
```

Comment:
Better use "if" install of "?" ternary operator

Reply:
fix that